File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.js

function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

function LinkedList(head) {
  this.head = head;
  this.tail = null;
}
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */
var mergeTwoLists = function(list1, list2) {
  const listArray = [];
  let currentNode = new ListNode();
  const resultList = new LinkedList(currentNode);
  let currentList1 = list1;
  let currentList2 = list2;
  let index = 0;

  while (currentList1) {
    listArray.push(currentList1.val);
    currentList1 = currentList1.next;
  }

  while (currentList2) {
    listArray.push(currentList2.val);
    currentList2 = currentList2.next;
  }
  
  listArray.sort((a, b) => a - b);

 while (index < listArray.length) {
  currentNode.next = new ListNode(listArray[index]);
  currentNode = currentNode.next;
  index++;
 }

return resultList.head.next;
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        current_node1 = list1
        current_node2 = list2
        merge_sorted_node_head = ListNode()
        merge_sorted_node = merge_sorted_node_head

        while current_node1 or current_node2:
            if not current_node1:
                merge_sorted_node.next = current_node2
                break
              
            if not current_node2:
                merge_sorted_node.next = current_node1
                break

            if current_node1.val >= current_node2.val:
                merge_sorted_node.next = current_node2
                current_node2 = current_node2.next
            else:
                merge_sorted_node.next = current_node1
                current_node1 = current_node1.next
            
            merge_sorted_node = merge_sorted_node.next
        
        return merge_sorted_node_head.next